### PR TITLE
Normative: `AsyncFromSyncIterator` is exposed to user code

### LINKED ISSUE #66: Revert type assertions in %AsyncFromSyncIteratorPrototype% methods
https://tc39.es/ecma262/#sec-%asyncfromsynciteratorprototype%.next

These methods used to throw TypeError if **this** was the wrong type, but someone must have noticed that these objects are never exposed to user code, and the type checks in the spec were changed to assertions.

However, `AsyncIterator.from` will create a way to expose these objects to user code, so that their methods can be misapplied. I think this proposal needs to reinstate the type checks... assuming these methods aren't [removed from ECMA-262 entirely](https://github.com/tc39/ecma262/issues/1765).

### PR DESCRIPTION
**Pull request by ExE-Boss**
Resolves #66

### CODE REVIEW

**Comment by codehag** [comment]
This patch might be out of date, what do you think @devsenk?

**Comment by avandolder** [comment]
I think you meant @devsnek :)

**Comment by codehag** [comment]
thank you @avandolder 😅

File: spec.html [hunk 1]
```diff
@@ -756,5 +756,76 @@ <h1>%AsyncIterator.prototype% [ @@toStringTag ]</h1>
...
+    <emu-clause id="sec-%asyncfromsynciteratorprototype%-object">
+      <h1>The %AsyncFromSyncIteratorPrototype% Object</h1>
+
+      <emu-clause id="sec-%asyncfromsynciteratorprototype%.next">
+        <h1>%AsyncFromSyncIteratorPrototype%.next ( _value_ )</h1>
+        <emu-alg>
+          1. Let _O_ be the *this* value.
+          1. <del>Assert: Type(_O_) is Object and _O_ has a [[SyncIteratorRecord]] internal slot.</del>
+          1. Let _promiseCapability_ be ! NewPromiseCapability(%Promise%).
+          1. <ins>If Type(_O_) is not Object, or if _O_ does not have a [[SyncIteratorRecord]] internal slot, then</ins>
```
Comment (devsnek) [inline spec.html:769]: you can use RequireInternalSlot instead.

Comment (ExE-Boss) [reply spec.html:769]: That will cause the function to throw a `TypeError` instead of returning a rejected `Promise`.

Comment (ljharb) [reply spec.html:769]: IfAbruptRejectPromise may be useful here

**Pull request merged**